app/dot/cpp/core/services/EntityService.java

```diff
@@ -92,7 +105,8 @@ public void delete(T entity) {
     repository.delete(entity);
   }
 
-  public <S extends BaseRequest> S getRequest(String id, S request, BiConsumer<S, T>... consumers) {
+  public <S extends BaseRequest> S getRequest(String id, S request, BiConsumer<S, T>... consumers)
+      throws EntityNotFoundException {
```

Comment:
this method first checks the id, the same checks happen again in findById, we could surround findById in a try-catch and on catch return the request